tests/Neo.Plugins.DBFTPlugin.Tests/ConsensusTestHelper.cs

```diff
@@ -0,0 +1,303 @@
+        }
+
+        /// <summary>
+        /// Creates a PrepareResponse message
+        /// </summary>
+        public PrepareResponse CreatePrepareResponse(UInt256 preparationHash = null)
+        {
+            return new PrepareResponse
+            {
+                PreparationHash = preparationHash ?? UInt256.Zero
+            };
+        }
+
+        /// <summary>
+        /// Creates a Commit message
+        /// </summary>
+        public Commit CreateCommit(byte[] signature = null)
+        {
+            return new Commit
+            {
+                Signature = signature ?? new byte[64] // Fake signature for testing
+            };
+        }
+
+        /// <summary>
+        /// Creates a ChangeView message
+        /// </summary>
+        public ChangeView CreateChangeView(ChangeViewReason reason = ChangeViewReason.Timeout)
+        {
+            return new ChangeView
+            {
+                Timestamp = (ulong)DateTimeOffset.UtcNow.ToUnixTimeMilliseconds(),
+                Reason = reason
+            };
+        }
+
+        /// <summary>
+        /// Creates a RecoveryRequest message
+        /// </summary>
+        public RecoveryRequest CreateRecoveryRequest()
+        {
+            return new RecoveryRequest
+            {
+                Timestamp = (ulong)DateTimeOffset.UtcNow.ToUnixTimeMilliseconds()
+            };
+        }
+
+        /// <summary>
+        /// Sends a message to multiple consensus services
+        /// </summary>
+        public void SendToAll(ExtensiblePayload payload, IActorRef[] consensusServices)
+        {
+            foreach (var service in consensusServices)
+            {
+                service.Tell(payload);
+            }
+        }
+
+        /// <summary>
+        /// Sends a message to specific consensus services
+        /// </summary>
+        public void SendToValidators(ExtensiblePayload payload, IActorRef[] consensusServices, int[] validatorIndices)
+        {
+            foreach (var index in validatorIndices)
+            {
+                if (index >= 0 && index < consensusServices.Length)
+                {
+                    consensusServices[index].Tell(payload);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Simulates a complete consensus round
+        /// </summary>
+        public void SimulateCompleteConsensusRound(IActorRef[] consensusServices, uint blockIndex = 1, UInt256[] transactions = null)
+        {
+            var validatorCount = consensusServices.Length;
+            var primaryIndex = (int)(blockIndex % (uint)validatorCount);
+
+            // Step 1: Primary sends PrepareRequest
+            var prepareRequest = CreatePrepareRequest(transactionHashes: transactions);
+            var prepareRequestPayload = CreateConsensusPayload(prepareRequest, primaryIndex, blockIndex);
+            SendToAll(prepareRequestPayload, consensusServices);
+
+            // Step 2: Backup validators send PrepareResponse
```

Comment:
@Jim8y , here, in the past, what we did is to wait for the receivers of the last `SendToAll(prepareRequestPayload, consensusServices);` to trigger the next PrepareResponse. So, we listen to their IActorRef to see when they receive.

Reply:
I've created comprehensive tests in UT_DBFT_ProperMessageFlow.cs that follow your recommended approach